### DISCUSSION TITLE
add a annotation with custom text when you click on a model

### DISCUSSION THREAD

**Discussion by sietsenefs** [google/model-viewer#2233] [Q&A]
I know this is already possible somehow since it is used here: https://modelviewer.dev/editor/ at the edit icon then add hotspot button but I cant figure out how to reverse engineer the code used because of so many files referenced in the JavaScript to make it work. I also can't find someone else who managed to make it work although I have seen a few posts of people wanting to add this but they also did not manage to do it. 

Is there something I overlooked where this is maybe explained? or is my only option to keep trying to reverse engineer the code used in the editor example?

**Comment by elalish** [comment]
This function is the key: https://modelviewer.dev/docs/#entrydocs-annotations-methods-positionAndNormalFromPoint

It allows you to convert a user click into the position and normal that you can then add as the hotspot's attributes.
